dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/compute/function/DatacenterToLocation.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.dimensiondata.cloudcontrol.compute.function;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.jclouds.dimensiondata.cloudcontrol.domain.Datacenter;
+import org.jclouds.domain.Location;
+import org.jclouds.domain.LocationBuilder;
+import org.jclouds.domain.LocationScope;
+import org.jclouds.location.suppliers.all.JustProvider;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import static com.google.inject.internal.util.$Iterables.getOnlyElement;
+import static com.google.inject.internal.util.$Preconditions.checkNotNull;
+
+@Singleton
+public class DatacenterToLocation implements Function<Datacenter, Location> {
+
+   private final JustProvider justProvider;
+
+   @Inject
+   public DatacenterToLocation(JustProvider justProvider) {
+      this.justProvider = checkNotNull(justProvider, "justProvider");
+   }
+
+   @Override
+   public Location apply(final Datacenter datacenter) {
+      return new LocationBuilder().id(datacenter.id()).description(datacenter.displayName())
+            .parent(getOnlyElement(justProvider.get())).scope(LocationScope.ZONE)
```

Comment:
The location hierarchy is usually "provider > region > zone". Should DimensionData locations be considered regions instead of zones? Otherwise, why the gap?

Reply:
In DD we do have the concept of regions, but for each of the regions we will have a different API endpoint. When we use DD from within jclouds we will operate on a single region only. Within each region we will have multiple zones. So in my view a Datacenter is equivalent to a Zone.

Reply:
Can the same credentials be configured to access all available regions?
Are the region endpoints fixed? (Something we can preconfigure in the provider metadata properties)? If so, I expand a bit on how to properly configure this, in a similar way we do with `aws-ec2` (there are there also different endpoints for each region).

Reply:
@nacx we can use the same credentials for the different regions.

I see how the regions and end points are configured in aws-cloudwatch [here](https://github.com/trevorflanagan/jclouds/blob/8bbcfed312f6a19cd89441267f081c900b478c2c/providers/aws-cloudwatch/src/main/java/org/jclouds/aws/cloudwatch/AWSCloudWatchProviderMetadata.java#L56). Is this what you are talking about?

I can see some specific region setup in EC2 [here](https://github.com/trevorflanagan/jclouds/blob/8bbcfed312f6a19cd89441267f081c900b478c2c/providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/AWSEC2ProviderMetadata.java#L69)

Do you see these changes as something we should do for this PR or a different one?